spec/unit/pops/types/p_object_type_spec.rb

```diff
@@ -687,6 +675,35 @@ def parse_object(name, body_string)
       expect(eval_and_collect_notices(code)).to eql(['true'])
     end
 
+    it 'a named object is added to the loader irrespective of alias declaration' do
+      code = <<-CODE
+      $x = Object[{ name => 'MyFirstObject' }]
+      notice($x == MyFirstObject)
+      CODE
```

Comment:
I'm fine with changing so that no binding is made by the name used in the hash, but we need to sort out what should happen with both names are given:
```
type Foo = Object[{name => 'Bar'}]
```
Should this be allowed at all? And if so, which name should be used in error messages? When do names conflict? It will get mildly confusing to write:
```
type Foo = Object[{
  name => 'Bar',
  attributes => {
    'id' => { type => Integer, final => true }
  }
}]
```
and then, in some other file:
```
type Bar = Object[{ parent => Foo
  attributes => {
    'id' => { type => Integer[0], override => true }
  }
}]
```
and have the parser tell you `Bar[id]` attempts to override a final attribute in `Bar`. With the code as it stands right now, this can never happen since the loader would raise an 'attempt to redefine Bar' exception.